### DISCUSSION TITLE
Animation Stack.Screen slide_from_left in IOS

### DISCUSSION THREAD

**Discussion by MC-HaoPhan** [react-navigation/react-navigation#11641] [Q&A]
Hi everyone, I want to create a left-to-right animation, but the documentation says that `slide_from_left` only works on Android. Does anyone have a solution?

**Comment by MarcusDelvecchio** [comment]
Having the same issue. According to [this](https://reactnavigation.org/docs/stack-navigator/#animation-related-options), changing **gestureDirection** in options will also change the animation direction the direction is flipped. But this also doesn't seem to be working for me on v6.

Edit:
Seems to be an issue with Native Stack Navigator, which I am using. Have implemented workaround explained [here](https://stackoverflow.com/a/74100819/13665910), where I specify `options={{ headerShown: false, animationTypeForReplace: "pop"}}` in the`<Stack.Screen/>` (within the **Native** stack navigator) and updated my corresponding navigation.navigate() calls to navigation.replace(). Works for me on iOS.

**Reply by MC-HaoPhan** [reply]
<img width="800" alt="Screenshot 2023-11-06 at 5 13 21 PM" src="https://github.com/react-navigation/react-navigation/assets/105186769/40359f86-1652-4bc9-ba55-2a791e729599">

Can you send me an example, I tried it and it doesn't work.

**Reply by MarcusDelvecchio** [reply]
Sorry I just realized that although this workaround allows for a **slide-from-left on open** and **slide-to-left on close** behaviour for the screen, the gestures do not work. Here is my solution.

<img width="1164" alt="image" src="https://github.com/react-navigation/react-navigation/assets/49920628/eb7be398-731c-42e3-972b-1f4561b3cf64">

and when navigating to the above screen so that it slides from left rather than from the right, I use navigation.replace("Explore"). In doing this, the current screen in the stack will be replaced by our new screen, and it will appear from the left. The issues with the gestures arise here however because since we are replacing instead of pushing this new screen when going from screen1 to screen2, we cannot use gestures to go back.

I tried to mess around with a custom navigation.dispatch() solution that will push the same screen1 to the stack at the same time we perform the navigation.replace, so that way we could use gestures to return, but I was not able to come up with a working solution and am simply going to move on. Please let me know if you make any progress in coming up with a work around.
